### DISCUSSION TITLE
clangd cannot figure C++ standard in STL headers in VS Code

### DISCUSSION THREAD

**Discussion by EienMiku** [clangd/clangd#2220] [Q&A] [answered]
I am using xmake and the clang toolchain with MSVC STL. I have set the standard to `-std=c++2c` and added this compilation option in the `.clangd` file. My `compile_commands.json`, which is generated by xmake, is as follows:
``` json
[
{
  "directory": "d:\\Coding\\Trails\\test\\test",
  "arguments": ["clang.exe", "-c", "-Qunused-arguments", "-m64", "-fvisibility=hidden", "-fvisibility-inlines-hidden", "-O3", "-fexceptions", "-fcxx-exceptions", "-std=c++2c", "-DNDEBUG", "-o", "build\\.objs\\test\\windows\\x64\\release\\src\\main.cpp.obj", "src\\main.cpp"],
  "file": "src\\main.cpp"
}]

```
## **In workspace cpp files**
Everything works fine, and the standard could be figured:
<img src="https://github.com/user-attachments/assets/736bd36f-5b5e-47d2-9390-1816233bb88b" width="400">

## Viewing STL standard library files through go-to-def
The standard seems to be the default C++14, as `_cplusplus` is `201402L`, and all macros for standards above C++14 are `0`:
<img src="https://github.com/user-attachments/assets/3550b25e-c910-4207-8011-95c7bfb1ad23" width="600">

I found a similar issue in the discussion, but it was related to the incompatibility of `cl.exe` with compile option `-std:c++latest`.
I have tried many methods, but clangd still cannot correctly recognize the C++ standard in the standard library.

I would appreciate any guidance or suggestions on how to resolve this issue. Thank you!

**Comment by HighCommander4** [comment, marked as answer]
Please see the explanation and solution at https://clangd.llvm.org/faq#how-do-i-fix-errors-i-get-when-opening-headers-outside-of-my-project-directory

**Reply by EienMiku** [reply]
Thank you!
